DQMOffline/Ecal/plugins/ECALMultifitAnalyzer_HI.cc

```diff
@@ -0,0 +1,293 @@
+    double phi = geomH->getGeometry(hit->detid())->getPosition().phi();
+    ee_chi2_eta->Fill(eta, hit->chi2() );
+    ee_errors_eta->Fill(eta, hit->energyError() );
+    if(hit->energy() > mRechitEnergyThreshold)
+    {
+      ee_chi2_e5->Fill(hit->chi2() );
+      ee_errors_e5->Fill(hit->energyError() );
+      ee_chi2_e5_eta->Fill(eta, hit->chi2() );
+      ee_errors_e5_eta->Fill(eta, hit->energyError() );
+    }
+
+    for (std::vector<reco::Photon>::const_iterator pho = recoPhotonsHandle->begin(); pho != recoPhotonsHandle->end(); ++pho) {
+      if(pho->et() < mRecoPhotonPtThreshold ) continue;
+      double dr = reco::deltaR(eta, phi, pho->eta(), pho->phi());
+      if(dr < mDeltaRPhotonThreshold)
+      {
+	ee_chi2_photon15->Fill(hit->chi2() );
+	ee_errors_photon15->Fill(hit->energyError() );
+      }
+    }
+    for (std::vector<reco::CaloJet>::const_iterator jet = recoJetHandle->begin(); jet != recoJetHandle->end(); ++jet) {
+      if(jet->pt() < mRecoJetPtThreshold ) continue;
+      double dr = reco::deltaR(eta, phi, jet->eta(), jet->phi());
+      if(dr < mDeltaRJetThreshold)
+      {
+	ee_chi2_jet30->Fill(hit->chi2() );
+	ee_errors_jet30->Fill(hit->energyError() );
+      }
+    }
+  }
+}
+
+
+// ------------ method called once each job just before starting event loop  ------------
+void
+ECALMultifitAnalyzer_HI::bookHistograms(DQMStore::IBooker& iBooker, edm::Run const&, edm::EventSetup const&)
+{
+
+  iBooker.setCurrentFolder ("EcalCalibration/HIN_MultiFitAnalyzer");
+
+  eb_chi2 = 0;
+  eb_chi2_eta = 0;
+  eb_chi2_e5 = 0;
+  eb_chi2_e5_eta = 0;
+  eb_errors = 0;
+  eb_errors_eta = 0;
+  eb_errors_e5 = 0;
+  eb_errors_e5_eta = 0;
+  eb_chi2_photon15 = 0;
+  eb_errors_photon15 = 0;
+  eb_chi2_jet30 = 0;
+  eb_errors_jet30 = 0;
+
+  ee_chi2 = 0;
+  ee_chi2_eta = 0;
+  ee_chi2_e5 = 0;
+  ee_chi2_e5_eta = 0;
+  ee_errors = 0;
+  ee_errors_eta = 0;
+  ee_errors_e5 = 0;
+  ee_errors_e5_eta = 0;
+  ee_chi2_photon15 = 0;
+  ee_errors_photon15 = 0;
+  ee_chi2_jet30 = 0;
+  ee_errors_jet30 = 0;
+
+  const int nBins = 500;
+  const float maxChi2 = 70;
+  const float maxError = 0.5;
+
+  TH2F * hProfile_Chi2 = new TH2F("hProfile_Chi2","",nBins, -5, 5, nBins, 0, maxChi2);
+  TH2F * hProfile_Err = new TH2F("hProfile_Err","",nBins, -5, 5, nBins, 0, maxError);
+
+  eb_chi2 = iBooker.book1D("rechit_eb_chi2","Rechit eb_chi2;chi2 fit value;",nBins,0,maxChi2);
+  eb_chi2_eta = iBooker.book2D("rechit_eb_eta_Vs_mean_Chi2", hProfile_Chi2);
+  eb_chi2_e5 = iBooker.book1D(Form("rechit_eb_chi2_e%2.0f", mRechitEnergyThreshold),Form("Rechit eb_chi2, e>%2.0fGeV;chi2 fit value;", mRechitEnergyThreshold),nBins,0,maxChi2);
+  eb_chi2_e5_eta = iBooker.book2D(Form("rechit_eb_chi2_e%2.0f_eta", mRechitEnergyThreshold),hProfile_Chi2);
+  eb_errors = iBooker.book1D("rechit_eb_errors","Rechit eb_errors;error on the energy;",nBins,0,maxError);
+  eb_errors_eta = iBooker.book2D("rechit_eb_errors_eta",hProfile_Err);
```

Comment:
Hi Slava;
if this is not particularly harmful for a known reason, we should not 
change it in this PR.



On 19/11/15 13:21, Slava Krutelyov wrote:
>
> In DQMOffline/Ecal/plugins/ECALMultifitAnalyzer_HI.cc 
> <https://github.com/cms-sw/cmssw/pull/12484#discussion_r45333540>:
>
> > +  ee_chi2_jet30 = 0;
> > +  ee_errors_jet30 = 0;
> > +
> > +  const int nBins = 500;
> > +  const float maxChi2 = 70;
> > +  const float maxError = 0.5;
> > +
> > +  TH2F * hProfile_Chi2 = new TH2F("hProfile_Chi2","",nBins, -5, 5, nBins, 0, maxChi2);
> > +  TH2F * hProfile_Err = new TH2F("hProfile_Err","",nBins, -5, 5, nBins, 0, maxError);
> > +
> > +  eb_chi2 = iBooker.book1D("rechit_eb_chi2","Rechit eb_chi2;chi2 fit value;",nBins,0,maxChi2);
> > +  eb_chi2_eta = iBooker.book2D("rechit_eb_eta_Vs_mean_Chi2", hProfile_Chi2);
> > +  eb_chi2_e5 = iBooker.book1D(Form("rechit_eb_chi2_e%2.0f", mRechitEnergyThreshold),Form("Rechit eb_chi2, e>%2.0fGeV;chi2 fit value;", mRechitEnergyThreshold),nBins,0,maxChi2);
> > +  eb_chi2_e5_eta = iBooker.book2D(Form("rechit_eb_chi2_e%2.0f_eta", mRechitEnergyThreshold),hProfile_Chi2);
> > +  eb_errors = iBooker.book1D("rechit_eb_errors","Rechit eb_errors;error on the energy;",nBins,0,maxError);
> > +  eb_errors_eta = iBooker.book2D("rechit_eb_errors_eta",hProfile_Err);
>
> @deguio <https://github.com/deguio> what's a preferred way of booking 
> 2D histograms?
> book2D should let you book internally using arguments instead of 
> making a histogram first
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/12484/files#r45333540>.
>